crates/polars-ops/src/frame/join/general.rs

```diff
@@ -32,7 +32,13 @@ pub fn _finish_join(
     let suffix = get_suffix(suffix);
 
     for name in rename_strs {
-        df_right.rename(&name, &_join_suffix_name(&name, suffix))?;
+        let new_name = _join_suffix_name(&name, suffix);
+        df_right.rename(&name, new_name.as_str()).map_err(|_| {
+            polars_err!(Duplicate: "column with name '{}' already exists.\n\n\
```

Comment:
There should be no punctuation here (see https://github.com/pola-rs/polars/issues/10421)

```suggestion
            polars_err!(Duplicate: "column with name '{}' already exists\n\n\
```